File: Daily Temperatures - Leetcode 739/Daily Temperatures - Leetcode 739.java
import java.util.*;

public class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] answer = new int[n];
        Stack<int[]> stack = new Stack<>();

        for (int i = 0; i < n; i++) {
            int temp = temperatures[i];
            while (!stack.isEmpty() && stack.peek()[0] < temp) {
                int[] prev = stack.pop();
                answer[prev[1]] = i - prev[1];
            }
            stack.push(new int[]{temp, i});
        }
        return answer;
    }
}

File: Daily Temperatures - Leetcode 739/Daily Temperatures - Leetcode 739.cpp
#include <vector>
#include <stack>

class Solution {
public:
    std::vector<int> dailyTemperatures(std::vector<int>& temperatures) {
        int n = temperatures.size();
        std::vector<int> answer(n, 0);
        std::stack<std::pair<int, int>> stk;  // pair<temperature, index>

        for (int i = 0; i < n; ++i) {
            int temp = temperatures[i];
            while (!stk.empty() && stk.top().first < temp) {
                auto [stk_temp, stk_index] = stk.top();
                stk.pop();
                answer[stk_index] = i - stk_index;
            }
            stk.push({temp, i});
        }
        return answer;
    }
};
